api/request.go

```diff
@@ -138,3 +138,68 @@ func (s SliceRequest) toString() (string, error) {
 	str := string(out)
 	return str, nil
 }
+
+// Query for Horizon endpoints
+// @Description Query payload for horizon endpoint /horizon.
+type HorizonRequest struct {
+	RequestedResource
+
+	// Horizon / height-map
+	Horizon [][]float32 `json:"horizon" binding:"required"`
+
+	// Rotation of the X-axis (East), counter-clockwise, in degrees
+	Rotation float32 `json:"rotation" binding:"required"`
+
+	// X-coordinate of the origin
+	Xori float32 `json:"xori" binding:"required"`
+
+	// Y-coordinate of the origin
+	Yori float32 `json:"yori" binding:"required"`
+
+	// X-increment - The physical distance between columns in horizon
+	Xinc float32 `json:"xinc" binding:"required"`
+
+	// Y-increment - The physical distance between rows in horizon
+	Yinc float32 `json:"yinc" binding:"required"`
+
+	// Any sample in the input horizon with value == fillValue will be ignored
+	// and the fillValue will be used in the amplitude map.
+	// I.e. for any [i, j] where horizon[i][j] == fillValue then
+	// output[i][j] == fillValue.
```

Comment:
We should also mention that fill value will be used for points that fall outside the seismic cube as well